processing/src/main/java/io/druid/query/groupby/GroupByQueryQueryToolChest.java

```diff
@@ -158,14 +160,27 @@ public GroupByQueryQueryToolChest(
       }
 
       final Sequence<Row> subqueryResult = mergeGroupByResults(subquery, runner, context);
-      final List<AggregatorFactory> aggs = Lists.newArrayList();
+      final Set<AggregatorFactory> aggs = Sets.newHashSet();
+
       for (AggregatorFactory aggregatorFactory : query.getAggregatorSpecs()) {
-        aggs.addAll(aggregatorFactory.getRequiredColumns());
+        for (final AggregatorFactory transferAgg : aggregatorFactory.getRequiredColumns()) {
```

Comment:
@xvrl I added a comment, hopefully it clears things up. Thanks for the feedback X.